client/state/jetpack-connect/test/selectors.js

```diff
@@ -434,4 +435,43 @@ describe( 'selectors', () => {
 			expect( getUserAlreadyConnected( trueState ) ).toBe( true );
 		} );
 	} );
+
+	describe( '#getPartnerIdFromQuery', () => {
+		test( 'should return null when no argument', () => {
+			expect( getPartnerIdFromQuery() ).toBeNull();
+		} );
+
+		test( 'should return null if no partner_id query present', () => {
+			const state = {
+				ui: {
+					route: {
+						query: {
+							initial: {
+								email_address: 'user@wordpress.com',
+							},
+						},
+					},
+				},
+			};
+			expect( getPartnerIdFromQuery( state ) ).toBeNull();
+		} );
+
+		test( 'should return partner ID as integer when partner_id query present', () => {
+			const state = {
+				ui: {
+					route: {
+						query: {
+							initial: {
+								email_address: 'user@wordpress.com',
+								partner_id: 49640,
+							},
+						},
+					},
+				},
+			};
+
+			// toBe is a strict equality check here.
+			expect( getPartnerIdFromQuery( state ) ).toBe( 49640 );
+		} );
+	} );
```

Comment:
It would be good to test a case where the `partner_id` from the query isn't a nice integer.